### PR TITLE
Use different ports in testing to enable parallel tests

### LINKED ISSUE #8: Fix unit tests port binding issue
https://github.com/knative-sandbox/control-protocol/runs/2011732092#step:6:50

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

Fix #8

### CODE REVIEW

**Comment by slinkydeveloper** [comment]
cc @devguyio

File: pkg/reconciler/connection_pool_test.go [hunk 1]
```diff
@@ -64,9 +65,9 @@ func mockValueMerger(old interface{}, new interface{}) interface{} {
 	return &merged
 }
 
-var serverConnectionPoolSetupTestCases = map[string]func(t *testing.T, ctx context.Context, opts ...ControlPlaneConnectionPoolOption) (control.Service, *ControlPlaneConnectionPool){
+var serverConnectionPoolSetupTestCases = map[string]func(t *testing.T, ctx context.Context, opts ...ControlPlaneConnectionPoolOption) (*network.ControlServer, *ControlPlaneConnectionPool){
```
Comment (devguyio) [inline pkg/reconciler/connection_pool_test.go:68]: NIT: maybe later as part of a separate PR, having a var in a file between funcs just triggers me badly :smile: maybe move it to the top of the file?

**Review by devguyio — APPROVED** [review]
Just a small NIT.
/lgtm

File: pkg/reconciler/connection_pool_test.go (hunk 1 above)
Comment (slinkydeveloper) [reply pkg/reconciler/connection_pool_test.go:68]: I want to burn these tests to the ground TBH :smile: I'm going to reorganize all these e2e tests in a separate package (maybe the `test` package introduced here https://github.com/knative-sandbox/control-protocol/pull/21). So we can have proper e2e tests without doing these bad tricks and a copy pasting good part of the initialization code

**Pull request merged**